# GCaMP+jRGECO(1)
file path : /Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/

In [7]:
import pandas as pd

input_path = "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_1_Subt2__0__81198.csv"
output_path = "/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_1_Subt2__0__81198.csv"

df = pd.read_csv(input_path)

# wavelength 추출
wavelengths = df.columns[3:].astype(float)

# intensity matrix만 추출
spectra = df.iloc[:, 3:].apply(pd.to_numeric, errors="coerce")

# 450–700 nm만 사용
mask = (wavelengths >= 450) & (wavelengths <= 700)

mixed = spectra.loc[:, df.columns[3:][mask]]

# index를 Time1, Time2...로 변경
mixed.index = [f"Time{i+1}" for i in range(len(mixed))]

# 저장
mixed.to_csv(output_path)

print(mixed.shape)
print(mixed.head())
print(f"저장 완료: {output_path}")

(1177, 733)
       450.154  450.512  450.87  451.229  451.587  451.945  452.303  452.662  \
Time1     1.41     1.62    1.57     1.57     1.51     1.13     1.34     1.70   
Time2    -6.80    -6.66   -6.57    -6.79    -6.44    -6.39    -6.33    -6.26   
Time3     3.90     3.98    4.13     4.07     3.15     3.41     3.46     3.95   
Time4     5.34     5.28    5.43     5.15     5.34     5.33     5.05     5.69   
Time5     3.79     4.15    4.44     4.51     4.02     4.08     3.87     4.15   

       453.02  453.378  ...  696.922  697.245  697.568  697.891  698.214  \
Time1    1.82     2.25  ...    -1.28    -1.56    -1.49    -1.30    -1.72   
Time2   -6.07    -5.85  ...    -9.87   -10.36    -9.72    -9.44    -9.25   
Time3    4.08     4.57  ...     2.00     1.02     1.49     1.69     1.67   
Time4    5.75     6.11  ...     2.18     1.84     2.18     2.33     2.18   
Time5    4.62     5.39  ...    -1.95    -2.66    -2.59    -2.25    -2.80   

       698.537  698.86  699.183  699.506  699.829 

In [8]:
mixed.head()

,450.154,450.512,450.87,451.229,451.587,451.945,452.303,452.662,453.02,453.378,...,696.922,697.245,697.568,697.891,698.214,698.537,698.86,699.183,699.506,699.829
Time1,1.41,1.62,1.57,1.57,1.51,1.13,1.34,1.70,1.82,2.25,...,-1.28,-1.56,-1.49,-1.30,-1.72,-1.44,-1.23,-0.59,-0.80,-1.21
Time2,-6.80,-6.66,-6.57,-6.79,-6.44,-6.39,-6.33,-6.26,-6.07,-5.85,...,-9.87,-10.36,-9.72,-9.44,-9.25,-8.75,-8.95,-8.38,-8.39,-8.48
Time3,3.90,3.98,4.13,4.07,3.15,3.41,3.46,3.95,4.08,4.57,...,2.00,1.02,1.49,1.69,1.67,2.02,2.23,2.44,2.51,2.16
Time4,5.34,5.28,5.43,5.15,5.34,5.33,5.05,5.69,5.75,6.11,...,2.18,1.84,2.18,2.33,2.18,2.52,2.74,3.31,3.30,3.15
Time5,3.79,4.15,4.44,4.51,4.02,4.08,3.87,4.15,4.62,5.39,...,-1.95,-2.66,-2.59,-2.25,-2.80,-2.67,-2.26,-1.97,-2.54,-2.67


In [9]:
mixed.columns[:20]

Index(['450.154', '450.512', '450.87', '451.229', '451.587', '451.945',
       '452.303', '452.662', '453.02', '453.378', '453.736', '454.094',
       '454.452', '454.809', '455.167', '455.525', '455.883', '456.241',
       '456.598', '456.956'],
      dtype='object')

col 이 450, 451, 452 이렇게 있는게 아니라 파장값이 완전 소수점으로 되어 있음 \
근데 어쨌든 레퍼런스랑 동일해서 상관은 없을듯 ?

## file 경로
### input path
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_2_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_3_Subt2__0__88268.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_4_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_5_Subt2__0__88268.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_6_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_7_Subt2__0__88268.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_8_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_9_Subt2__0__88268.csv \
/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_10_Subt2__0__81198.csv 
### ouput path
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_2_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_3_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_4_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_5_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_6_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_7_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_8_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_9_Subt2__0__81198.csv \
/Users/rnlghdb/NDL/data/unmixing_spectrum/Flna29_gr25uw_10_Subt2__0__81198.csv


In [11]:
import os
import pandas as pd

input_files = [
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_2_Subt2__0__81198.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_3_Subt2__0__88268.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_4_Subt2__0__92821.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_5_Subt2__0__96263.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_6_Subt2__0__99471.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_7_Subt2__0__102569.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_8_Subt2__0__107144.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_9_Subt2__0__111258.csv",
    "/Users/rnlghdb/NDL/data/GCaMP+jRGECO(1)/Flna29_gr25uw_10_Subt2__0__115761.csv"
]

output_dir = "/Users/rnlghdb/NDL/data/unmixing_spectrum"
os.makedirs(output_dir, exist_ok=True)

for input_path in input_files:

    df = pd.read_csv(input_path)

    # wavelength 추출
    wavelengths = df.columns[3:].astype(float)

    # intensity matrix
    spectra = df.iloc[:, 3:].apply(
        pd.to_numeric,
        errors="coerce"
    )

    # 450–700 nm
    mask = (
        (wavelengths >= 450) &
        (wavelengths <= 700)
    )

    mixed = spectra.loc[
        :,
        df.columns[3:][mask]
    ]

    # Time1 ~ TimeN
    mixed.index = [
        f"Time{i+1}"
        for i in range(len(mixed))
    ]

    # 파일명 유지
    output_path = os.path.join(
        output_dir,
        os.path.basename(input_path)
    )

    mixed.to_csv(output_path)

    print(
        f"완료: {os.path.basename(input_path)} "
        f"{mixed.shape}"
    )

print("모든 파일 저장 완료")

완료: Flna29_gr25uw_2_Subt2__0__81198.csv (1177, 733)
완료: Flna29_gr25uw_3_Subt2__0__88268.csv (1177, 733)
완료: Flna29_gr25uw_4_Subt2__0__92821.csv (1177, 733)
완료: Flna29_gr25uw_5_Subt2__0__96263.csv (1177, 733)
완료: Flna29_gr25uw_6_Subt2__0__99471.csv (1177, 733)
완료: Flna29_gr25uw_7_Subt2__0__102569.csv (1177, 733)
완료: Flna29_gr25uw_8_Subt2__0__107144.csv (1177, 733)
완료: Flna29_gr25uw_9_Subt2__0__111258.csv (1177, 733)
완료: Flna29_gr25uw_10_Subt2__0__115761.csv (1177, 733)
모든 파일 저장 완료
